addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js

```diff
@@ -181,11 +182,12 @@ export class Powerbox {
                 ? this._currentOpenOptions.valueChangeFunction(term)
                 : this._filter(term, this._currentOpenOptions.commands);
 
+        const selection = this.options.document.getSelection();
         const showOnceOnKeyup = () => {
             this.show();
             openOnKeyupTarget.removeEventListener('keyup', showOnceOnKeyup, true);
-            initialTarget = openOnKeyupTarget;
-            this._initialValue = openOnKeyupTarget.textContent;
+            initialTarget = selection.anchorNode;
+            this._initialValue = selection.anchorNode.textContent;
```

Comment:
Thanks for your explanation, it's now all clear to me. Now, what if `selection.anchorNode` is a text node that was split in pieces, like so: `"abc""[]d""ef"`? If we just type it seems ok but then say I press the right arrow key twicebefore typing some more (`"abc""/d""eCOMMAND[]f"`). What I typed here won't be taken into account in the diff right? Using `closestElement(selection.anchorNode)` might suffice but then we don't support the same case but where `e` is in a `<i>` tag for instance.
So I suggest using `closestBlock(selection.anchorNode)` instead.

cc @sebgeelen @Goaman for their opinions.

Reply:
`closestBlock` seems the right approach to me indeed. 
It would still need to be thoroughly tested to ensure it work as intended though.

Reply:
Taking initialTarget as `closestBlock` how about we close the powerbox if user moves out of that block using arrow keys?

Reply:
That would make sense yes!